### PR TITLE
Feat/cloudfront api gateway support

### PR DESCRIPTION
**Pull request by davidweterings**

### CODE REVIEW

File: src/mach/templates/partials/aws.tf [hunk 1]
```diff
@@ -34,6 +34,10 @@ provider "aws" {
   {% include 'partials/endpoints/aws_url_locals.tf' %}
 {% endif %}
 
+{% if aws.cloud_front %}
```
Comment (tleguijt) [inline src/mach/templates/partials/aws.tf:37]: Shall we name this `cloudfront`? More in line with how it's named in Terraform provider

Comment (davidweterings) [reply src/mach/templates/partials/aws.tf]: agreed!

File: src/mach/templates/partials/endpoints/aws_gateway_cloudfront.tf [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+resource "aws_acm_certificate" "cloudfront_api_gateway" {
+  domain_name       = {{ endpoint.url|tf }}
+  validation_method = "DNS"
+
+  provider = aws.mach-cf-us-east-1
+}
+
+resource "aws_route53_record" "cloudfront_api_gateway_acm_validation" {
```
Comment (tleguijt) [inline src/mach/templates/partials/endpoints/aws_gateway_cloudfront.tf:21]: The entire ACM certificate + validation + CNAME record don't have to be duplicated.
They both share the same configurations.
Maybe we can rename `aws_api_gateway` to `aws_endpoints`, set up certificate + validation records and API gateway from there

File: src/mach/templates/partials/endpoints/aws_gateway_cloudfront.tf [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  viewer_certificate {
+    acm_certificate_arn      = aws_acm_certificate.cloudfront_api_gateway.arn
+    ssl_support_method       = "sni-only"
+    minimum_protocol_version = "TLSv1"
+
+  }
+}
+
+resource "aws_route53_record" "cloudfront_domain" {
```
Comment (tleguijt) [inline src/mach/templates/partials/endpoints/aws_gateway_cloudfront.tf:96]: Resource should have distinct name based on the endpoint key. Right now it's not possible to have two endpoints with CDN

File: src/mach/templates/partials/endpoints/aws_gateway_cloudfront.tf (hunk 2 above)
Comment (davidweterings) [reply src/mach/templates/partials/endpoints/aws_gateway_cloudfront.tf]: done

File: src/mach/templates/partials/endpoints/aws_gateway_cloudfront.tf (hunk 3 above)
Comment (davidweterings) [reply src/mach/templates/partials/endpoints/aws_gateway_cloudfront.tf]: Fixed!

File: src/mach/templates/partials/endpoints/aws_endpoints.tf [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  depends_on = [
+    {% for component in endpoint.components %}
+    module.{{ component.name }},
+    {% endfor %}
+  ]
+}
+
+{% if endpoint.url %}
+  {% if endpoint.enable_cdn %}
+    provider "aws" {
```
Comment (tleguijt) [inline src/mach/templates/partials/endpoints/aws_endpoints.tf:37]: De provider wordt nu wel meerdere keren ge-init op het moment dat je meerdere endpoints met CDN hebt. Dat misschien ook bundelen (op de plek waar we alle providers aanmaken?)

Comment (davidweterings) [reply src/mach/templates/partials/endpoints/aws_endpoints.tf]: Fixed, maakt het idd overzichtelijker qua provider instanties

**Comment by tleguijt** [comment]
LGTM!

**Pull request merged**